File: hidden-api/src/main/java/android/content/pm/PackageParser.java
package android.content.pm;

import android.content.IntentFilter;

import java.io.File;
import java.util.ArrayList;

public class PackageParser {

  public Package parsePackage(File packageFile, int flags) throws PackageParserException {
    throw new RuntimeException("Stub");
  }

  public static class PackageParserException extends Exception {
  }

  public static abstract class Component<II extends IntentInfo> {
    public final ArrayList<II> intents = null;
    public final String className = null;
  }

  public static abstract class IntentInfo extends IntentFilter {

  }

  public final static class Package {
    public final ArrayList<Activity> activities = new ArrayList<Activity>(0);
    public final ArrayList<Activity> receivers = new ArrayList<Activity>(0);
    public final ArrayList<Service> services = new ArrayList<Service>(0);
  }

  public final static class Activity extends Component<ActivityIntentInfo> {

  }

  public final static class ActivityIntentInfo extends IntentInfo {

  }

  public final static class Service extends Component<ServiceIntentInfo> {

  }

  public final static class ServiceIntentInfo extends IntentInfo {

  }
}

File: hidden-api/src/main/java/android/content/pm/PackageParser.kt
package android.content.pm

import android.content.IntentFilter
import java.io.File

class PackageParser {

  @Throws(PackageParserException::class)
  fun parsePackage(packageFile: File, flags: Int): Package {
    throw RuntimeException("Stub")
  }

  class PackageParserException : Exception()

  abstract class Component<II : IntentInfo> {
    val intents: ArrayList<II>? = null
    val className: String? = null
  }

  abstract class IntentInfo : IntentFilter()

  class Package {
    val activities: ArrayList<Activity> = ArrayList(0)
    val receivers: ArrayList<Activity> = ArrayList(0)
    val services: ArrayList<Service> = ArrayList(0)
  }

  class Activity : Component<ActivityIntentInfo>()

  class ActivityIntentInfo : IntentInfo()

  class Service : Component<ServiceIntentInfo>()

  class ServiceIntentInfo : IntentInfo()
}